src/seamless_communication/models/inference/translator.py

```diff
@@ -120,16 +131,6 @@ def get_modalities_from_task(self, task: Task) -> Tuple[Modality, Modality]:
         else:
             return Modality.TEXT, Modality.SPEECH
 
-    @torch.no_grad()
-    def synthesize_speech(
-        self,
-        code: List[int],
-        lang: str,
-        speaker: Optional[int] = None,
-        dur_prediction: Optional[bool] = True,
-    ) -> Tuple[List[Tensor], int]:
-        return self.vocoder(code, lang, speaker, dur_prediction), self.sr
-
     @torch.no_grad()
```

Comment:
Not part of this PR, but you should prefer `@torch.inference_mode()` here that completely disables autograd.